tools/topology/topology2/include/components/tdfb.conf

```diff
@@ -0,0 +1,174 @@
+#
+#
+# A generic TDFB component. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.tdfb.attribute_name"
+#
+# Usage: this component can be used by declaring in the parent object. i.e.
+#
+# Object.Widget.tdfb."N" {
+#		index			1
+#	}
+# }
+
+#
+# Where M is pipeline ID and N is a unique integer in the parent object.
+
+<include/controls/mixer.conf>
+<include/controls/enum.conf>
+
+Class.Widget."tdfb" {
+	#
+	# Pipeline ID
+	#
+	DefineAttribute."index" {
+		type "integer"
+	}
+
+	#
+	# Unique instance for TDFB widget
+	#
+	DefineAttribute."instance" {
+		type "integer"
+	}
+
+	# Include common widget attributes definition
+	<include/components/widget-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+			"instance"
+		]
+		!mandatory [
+			"num_input_pins"
+			"num_output_pins"
+			"num_input_audio_formats"
+			"num_output_audio_formats"
+		]
+
+		!immutable [
+			"uuid"
+			"type"
+		]
+		!deprecated [
+			"preload_count"
+		]
+		unique	"instance"
+	}
+
+	#
+	# tdfb widget switch control
+	#
+	Object.Control {
+		# Switch controls
+                mixer."1" {
+                        Object.Base.channel.1 {
+                                name    "fc"
+                                shift	0
+                        }
+                        Object.Base.ops.1 {
+                                name	"ctl"
+                                info	"volsw"
+                                #259 binds the mixer control to switch get/put handlers
+                                get	259
+                                put	259
+                        }
+                        max 1
+                }
+                mixer."2" {
+                        Object.Base.channel.1 {
+                                name    "fc"
+                                shift	0
+                        }
+                        Object.Base.ops.1 {
+                                name	"ctl"
+                                info	"volsw"
+                                #259 binds the mixer control to switch get/put handlers
+                                get	259
+                                put	259
+                        }
+                        max 1
+                }
+
+		# Enum controls
+		enum."1" {
+			Object.Base {
+				channel.1 {
+					name	"fc"
+					reg	3
+					shift	0
+				}
+				text.0 {
+					name "mytext"
```

Comment:
@singalsu is this intentional to add 2 enum controls here with the same set of values? Also, can we rename this to something more meaningful like tdfb_enum_text ?

Reply:
I think I should add comment to explain all controls. The first enum control is for beam angle set, the second enum is for estimated sound arrival angle. I was wondering if I should get the values from an include file to avoid duplication. What do you think?